### PR TITLE
PAPP-23617 Update paramiko to use sha256

### PR DESCRIPTION
**Pull request by shiyingtu**
- Background: SOAR apps like SSH use paramiko for ssh
connection. However, paramiko does not support sha256 internally,
which results in `[digital envelope routines: EVP_DigestInit_ex] disabled for FIPS.`
error on FIPS enabled instance

- Fix: Fork from the orginal repo and allow sha256 instead of md5.
Future apps will point to this repo instead of the orginal repo

- Also fix static and semgrep tests

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 beautifulsoup4==4.9.1
-paramiko==2.7.2
+git+https://github.com/phantomcyber/paramiko.git@main #paramiko with sha256
```
Comment (shiyingtu) [inline requirements.txt:2]: Use custom paramiko from phantomcyber, which is forked from the original repo. In this way, it's easy to maintain and better for apps sharing similar issues.

**Comment by shiyingtu** [comment]
Static test failures: 

- Licenses. Fail since line 2 of requirements.txt does not satisfy pattern <module-name>=version. Track in PAPP-23692

**Review by gary-phantom — APPROVED** [review]
Tested the new code on FIPS ova running 5.2.0.77191

**Comment by gary-phantom** [comment]
Can we fix the license failure?

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,3 @@
 beautifulsoup4==4.9.1
-git+https://github.com/phantomcyber/paramiko.git@main #paramiko with sha256
+git+https://github.com/phantomcyber/paramiko.git@soar-v1.0 #paramiko=2.9.2
```
Comment (rgil-splunk) [inline requirements.txt:2]: can we make the tag something like `2.9.2-fips`? the `soar-v1.0` tag doesn't really indicate what the version / change is in itself and could be misinterpreted to be tied to some sort of v1.0 of SOAR

**Review by rgil-splunk — APPROVED** [review]

**Comment by rgil-splunk** [comment]
static check has been updated to support git syntax and re-run on the latest comment, however we'll need to add it to approved modules given the custom fork: 

"Package `paramiko` version `2.9.2-fips` has license `license not found` with status `package not found`"

**Comment by shiyingtu** [comment]
> static check has been updated to support git syntax and re-run on the latest comment, however we'll need to add it to approved modules given the custom fork:
> 
> "Package `paramiko` version `2.9.2-fips` has license `license not found` with status `package not found`"

Package paramiko has 'LGPL' license in its wheel METADATA. If approved, I could add it to our list licenses

**Pull request merged**